web/src/main/java/edu/softserve/zoo/web/security/TokenAuthFilter.java

```diff
@@ -0,0 +1,51 @@
+package edu.softserve.zoo.web.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.RememberMeServices;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Filter which intercepts unauthenticated requests for secured resources.
+ * Uses {@link RememberMeServices} that authenticates request if there is a valid authentication token in it.
+ *
+ * @author Ilya Doroshenko
+ */
+@Component
+public class TokenAuthFilter extends GenericFilterBean {
+
+    @Autowired
+    RememberMeServices rememberMeServices;
```

Comment:
We are extracting all handling logic from filter itself, so I think it's a great extension point to add our custom token support.